dali/operators/signal/fft/power_spectrum.cc

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/signal/fft/power_spectrum.h"
+#include "dali/core/static_switch.h"
+#include "dali/kernels/signal/fft/fft_cpu.h"
+#include "dali/pipeline/data/views.h"
+
+#define FFT_SUPPORTED_NDIMS (1, 2, 3)
```

Comment:
Why only these particular dims are supported? Is it kernel's restrictions?

Reply:
Yes, it is restricted at the kernels size. Also, this is what would (and a little beyond) make sense for PowerSpectrum. You are calculating the power spectrum on either a 1D signal, or 2D with the second dimension being the number of channels. 3 dims is actually quite unlikely but one might have the windows already extracted (or provided externally) and use the PowerSpectrum operator to generate the Spectrogram